### ISSUE TITLE
Feature request: cuddled multiline conditions

### ISSUE THREAD

**Issue by demerphq** [perltidy/perltidy#110]
I am not sure if this just my failure to comprehend the docs, or if there is no option for this, but i really would like to format multi-line conditions with the open and close parens on their own line, with the close parent "cuddled" with the open brace. Eg:

```
        while ( 
                $la_pr > $op_pr ||    # any and larger
                ( $is_right_assoc{ $op_token->{op} }
                  and $la_pr == $op_pr )    # right and equal
        ){
            my $new_precedence= $op_pr + ($la_pr > $op_pr ? 1 : 0);
            $rhs= $self->_parse_expr_assoc($rhs, $new_precedence);
            $la= $tokens->[0];
            $la_pr= $self->_precedence($la);
        }
```
Note I don't have strong preferences about the formatting of the code inside the parens, but if the parens ends up multiline I dont want it hanging weirdly. For instance with my current set of options for perltidy i end up with the follow, which has a weird parens setup:
```
        while (
            $la_pr > $op_pr ||    # any and larger
            (       $is_right_assoc{ $op_token->{op} }
                and $la_pr == $op_pr)    # right and equal
            )
        {
            my $new_precedence= $op_pr + ($la_pr > $op_pr ? 1 : 0);
            $rhs= $self->_parse_expr_assoc($rhs, $new_precedence);
            $la= $tokens->[0];
            $la_pr= $self->_precedence($la);
        }
```

Perhaps this is because of my chosen options:

```
perltidy -w -ple -bbb -bbc -bbs -nolq -l=80 -noll -nola -nwls='='    -isbc -nolc -otr -kis -ci=4 -se -sot -sct -nsbl -pt=2 -fs     -fsb='#start-no-tidy' -fse='#end-no-tidy'
```

Another case that I would love some control over is this:
```
eval {
        ($line, $cond)= $self->_normalize_if_elif($line);
        1;
    }
        or die sprintf "Error at line %d\nLine %d: %s\n%s",
        ($line_info->start_line_num()) x 2, $line, $@;
```

I would rather this looks like:

```
eval {
        ($line, $cond)= $self->_normalize_if_elif($line);
        1;
 } or die sprintf "Error at line %d\nLine %d: %s\n%s",
          ($line_info->start_line_num()) x 2, $line, $@;

```

Is this already possible and i am just using the wrong options to achieve it?

BTW, love the tool. Use it all the time. Thanks so much for your efforts on this!

**Comment by shancock9** [comment]
I have written tentative updates to do both of these. The first option would be turned on with the new flag ``--cuddled-paren-brace``.  This will cause perltidy to join two lines which otherwise would be

```
  ) 
{
```

into a single line

```
) {

```

It is worth mentioning that the ``-bar`` flag will do this in this particular example, but it is not the same because ``-bar`` will move the brace to the end of a line if that is where the closing paren is.

The second looks for lines which would otherwise be

```
}
   or ...
```
and joins them, where the ``or`` can be one of a number of logical operators or ``if unless``.  Perltidy already does this in some cases, but not always, so this has to be implemented as a flag with an integer value.  I am tentatively using ``--vertical-tightness-brace-follower=n``, where this would probably be ``n=2`` for this variation.

Thanks for the suggestion. I will add a tagged version for testing in a few days.

**Comment by demerphq** [comment]
> Thanks for the suggestion. I will add a tagged version for testing in a few days.

Cool. I am excited! I am also glad that I didn't waste your time with something that was in the manual. :-) Thanks a lot!

**Comment by shancock9** [comment]
The next release to CPAN with these new options is several months away, but tagged version 20221112.02 at

https://github.com/perltidy/perltidy/tags

has them and is available for testing.  The updates for these are fairly simple so I don't anticipate any problems. The two new options are not yet documented in the manual but are as follows:


**-cpb, or --cuddled-paren-brace**


This option  will cause perltidy to join two lines which otherwise would be, by default,

```
        )
      {
```
      into a single line
```
      ) {
```

**-bfvt=n, or --brace-follower-vertical-tightness=n**,

For n=2, this option looks for lines which would otherwise be, by default,

``` 
     }
        or ..
```

and joins them into a single line

```
      } or ..
```

where the ``or`` can be one of a number of logical operators or ``if unless``.

The default is not to do this and can be indicated with n=1.

**Comment by demerphq** [comment]
Just wanted to say ive played with the version you referenced and it seems to work exactly as expected. Thanks! Merry Xmas.

**Comment by shancock9** [comment]
Thanks for verifying that it works, and Merry Christmas! I will close this issue after the next release to CPAN.

**Comment by rabbiveesh** [comment]
Interesting.
I installed from CPAN, and indeed it's the correct version. But the POD for
`perltidy` does not show these options. Manually inspecting the file for
`Perl::Tidy` shows that they exist, though.

**Comment by shancock9** [comment]
Yes, it looks like the pod docs did not get updated for these new parameters with the above instructions. I'll update it next time, and leave this open.
Thanks for catching this.

**Comment by shancock9** [comment]
This is fixed in version 20230701 at CPAN
